lib/core/include/irods/irods_load_plugin.hpp

```diff
@@ -94,6 +101,274 @@ namespace irods
+                    {"plugin_path", _plugin_path},
+                    {"message", "Could not parse DT_FLAGS_1: invalid file format (ELF magic number mismatch)"}
+                });
+                // clang-format on
+                // NOLINTNEXTLINE(hicpp-signed-bitwise)
+                return rtld_lazy_or_now | rtld_local_or_global | rtld_nodelete;
+            }
+
+            // get ELFCLASS
+            std::uint8_t ei_class{};
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+            lib.read(reinterpret_cast<char*>(&ei_class), sizeof(ei_class));
+
+            lib.seekg(0);
+
+            std::uint64_t e_shoff{};
+            std::uint16_t e_shentsize{};
+            std::uint64_t e_shnum{};
+            std::uint32_t e_shstrndx{};
+            if (ei_class == ELFCLASS32) {
+                Elf32_Ehdr elf_hdr_in{};
+                // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+                lib.read(reinterpret_cast<char*>(&elf_hdr_in), sizeof(elf_hdr_in));
+                e_shoff = elf_hdr_in.e_shoff;
+                e_shentsize = elf_hdr_in.e_shentsize;
+                e_shnum = elf_hdr_in.e_shnum;
+                e_shstrndx = elf_hdr_in.e_shstrndx;
+            }
+            else if (ei_class == ELFCLASS64) {
+                Elf64_Ehdr elf_hdr_in{};
+                // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+                lib.read(reinterpret_cast<char*>(&elf_hdr_in), sizeof(elf_hdr_in));
+                e_shoff = elf_hdr_in.e_shoff;
+                e_shentsize = elf_hdr_in.e_shentsize;
+                e_shnum = elf_hdr_in.e_shnum;
+                e_shstrndx = elf_hdr_in.e_shstrndx;
+            }
+            else {
+                // clang-format off
+                logger::warn({
+                    {"plugin_path", _plugin_path},
+                    {"message", "Could not parse DT_FLAGS_1: unknown or invalid ELFCLASS"},
+                    {"ELFCLASS", std::to_string(static_cast<std::uint32_t>(ei_class))}
+                });
+                // clang-format on
+                // NOLINTNEXTLINE(hicpp-signed-bitwise)
+                return rtld_lazy_or_now | rtld_local_or_global | rtld_nodelete;
+            }
+
+            // TODO: some platforms do not require a section table for libraries
+            if (e_shoff == 0) {
+                // clang-format off
+                logger::warn({
+                    {"plugin_path", _plugin_path},
+                    {"message", "Could not parse DT_FLAGS_1: No section header table"}
+                });
+                // clang-format on
+                // NOLINTNEXTLINE(hicpp-signed-bitwise)
+                return rtld_lazy_or_now | rtld_local_or_global | rtld_nodelete;
+            }
+
+            // seek to section table
+            lib.seekg(e_shoff);
+
+            // some data may be contained in the first entry
+            if ((e_shnum == 0) || (e_shstrndx == SHN_XINDEX)) {
```

Comment:
For non-const variables, let's invert the operands on each side of `==` to guard against future modifications.
```c++
0 == e_shnum || ...
```

Reply:
You're gonna have to unwrap that one for me. How does switching the operands guard against future modifications?

Reply:
just protects against fat-fingering the `==` to an assignment...

https://en.wikipedia.org/wiki/Yoda_conditions

Reply:
That makes sense now that I think about it. I'll make the changes.